### ISSUE TITLE
Render order of material is broken

### ISSUE THREAD

**Issue by andrekoehler** [urho3d/Urho3D#2233]
Changing render order in materials seems to have an effect, but there is still lots of Z-fighting: 
![grafik](https://user-images.githubusercontent.com/22099714/34872983-8c595fa0-f793-11e7-8208-67973459fbef.png)

I created a super simple minimal example scene that contains two overlapping planes. Both use Techniques/NoTextureUnlit.xml
[TestScene.zip](https://github.com/urho3d/Urho3D/files/1625859/TestScene.zip)

Isn't the render order feature supposed to fix exactly that kind of issue?

**Comment by andrekoehler** [comment]
If anyone else is having these issues and is using the Direct3D 11 backend, also read this other issue: https://github.com/urho3d/Urho3D/issues/1029

In order to make coplanar surfaces render in a specific order, give the material of the surface that has to be in front a constant bias between (exclusive) 0 and around -0.5 because -1 seems to not work on Direct3D 11

**Comment by andrekoehler** [comment]
Another note for users: Materials ending with the name "Alpha" will have depthwrite set to false. That means that depth bias does not have an effect. Instead use Material::SetRenderOrder to define which material shall come up on top.

**Comment by andrekoehler** [comment]
I finally solved this issue for my project. In my 3D scene there is a bottom plane that contains multiple CPU-generated 2D elements like movement paths and rectangles. I wanted to have full control over the render order of these 2D elements. I had to modify the file CoreData\RenderPaths\Forward.xml which is loaded by default by the Urho3D engine and editor. In there after the line with "clear" and before the line with pass="base" I added the following line:
`<command type="scenepass" pass="bottom" vertexlights="true" metadata="alpha" />`
This creates a new render pass. All elements in this render path are rendered before the objects on top of the bottom plane. This works because the camera is always above the bottom plane and never below it. In order to use this render pass I had to copy the default techniques that I use like CoreData\Techniques\NoTextureUnlit.xml, NoTextureUnlitAlpha.xml, DiffUnlit.xml and DiffUnlitAlpha.xml. In all these passes I removed all passes except one and replaced the pass names "base" and "alpha" with the name of my new pass "bottom". I also had to set the "depthwrite" attribute to true and the "depthtest" attribute to false. depthwrite="true" ensures that 3D objects, that are stuck a little into the bottom, look properly cut off at the bottom plane. depthtest="false" ensures, that the all the 2D elements in the bottom plane do not exhibit Z-fighting because they all share the same depth values.
